### PR TITLE
remove devices related to a service when device=true option used (see #596)

### LINKED ISSUE #596: missing support for device=true in the delete service endpoint
According to the documentation of the Configuration API, when removing a service the following parameter should allow to remove all attached devices to the service:
* `device (optional, boolean)`. Default value is false. Remove devices in service/subservice. This parameter is not valid when Fiware-ServicePath is '/*' or '/#'.

### PR DESCRIPTION
**Pull request by chicco785**
@fgalan @dcalvoalonso 
this is an attempt to fix #596, but i need a some support in understanding why i get an ECONNECT error from the nock server when unregistration are sent to context broker.

i supposed that:

```
            contextBrokerMock = nock('http://192.168.1.1:1026')
                .matchHeader('fiware-service', 'TestService')
                .matchHeader('fiware-servicepath', '/testingPath')
                .post('/NGSI9/registerContext')
                .reply(200,
                utils.readExampleFile(
                   './test/unit/examples/contextAvailabilityResponses/unregisterDevice1Success.json'));
```

would have been the only thing needed, but apparently something is still wrong (when using a real context broker there is no issue).

thx.

### CODE REVIEW

**Comment by fgalan** [comment]
nock configuration could be tricky... if the request doesn't exactly match with what the mock expects, it fails without providing much more detail.

Anyway, in the case it helps, could you provide the exact ECONNECT error message you get?

**Comment by chicco785** [comment]
I realised that it is a tricky tool, the weird thing in this case is that it looks like the agent cannot connect to the nock (I had before errors where the request was not matching and i understand the difference). Not sure there is a time out by which the nock is killed. I will re-run the test and pass the error.

**Comment by fgalan** [comment]
> I will re-run the test and pass the error.

Not sure of understanding... you mean that the problem has been solved?

**Comment by chicco785** [comment]
i just meant that i need to extract the full stack error and pass it here :) sorry for the confusion

**Comment by chicco785** [comment]
88ce2be fixed the ECONNECT error, still to have the correct reply, i had to insert an empty call to the iot agent api, in the code, since (not sure why, maybe it was just a time issue) the registry delete operation was called after the call for checking the fact that the device where cancelled.

**Comment by chicco785** [comment]
i believe this is ready for merge

File: lib/services/groups/groupService.js [hunk 1]
```diff
@@ -27,6 +27,7 @@ var async = require('async'),
     iotManagerService = require('./../common/iotManagerService'),
```
Comment (fgalan) [inline lib/services/groups/groupService.js]: An entry in the CHANGES_NEXT_RELEASE regarding the changes introduced by this PR should be added (I guess that mentioning issue #596)

**Comment by fgalan** [comment]
> i believe this is ready for merge

A comment regarding CHANGES_NEXT_RELEASE seems to be the only thing pending (although I'd have a new look at the end, given it has passed some time :) A sync updating recent master would be also a good idea.

**Comment by chicco785** [comment]
Rebase for sure!

iFede - Fede's mobile edition

**Comment by dcalvoalonso** [comment]
Just one comment, shall we include also some unit tests for NGSIv2?

**Comment by chicco785** [comment]
currently device-api-group-test is not at all covered for NGSIv2 (i.e. also all the other tests have not been implemented for NGSIv2)

**Comment by dcalvoalonso** [comment]
Maybe you have to do a pull of master branch, I see some NGSIv2 tests for provisioning: https://github.com/telefonicaid/iotagent-node-lib/tree/master/test/unit/ngsiv2/provisioning

**Comment by chicco785** [comment]
hi @dcalvoalonso i had a quick look again, but it seems to me that none of the test covers device group provisioning.

**Comment by dcalvoalonso** [comment]
Ups, you're right @chicco785 !! Sorry, too much GitHub issues for a single day!! :P

**Comment by dcalvoalonso** [comment]
LGTM

**Comment by fgalan** [comment]
@chicco785 sorry for the delay in this PR. Currently we are giving priority to a task related with upgrading node version in the library but once we finish with it, this PR will be progressed.

**Comment by fgalan** [comment]
Merging. Thanks for the contribution!

**Pull request merged**